### DISCUSSION TITLE
One adapter, several ebus devices?

### DISCUSSION THREAD

**Discussion by cibernox** [john30/ebusd#615] [General]
Hi! I have bought one ebus to usb adapter (the ebuzzz adapter since the "official" ebus adapter has been out of stock for a long time and this is the only one I was able to get).
I just ran the wire from the server rack to the equipment room, I haven't yet setup any of this. I intend to connect the adapted directly to the raspberry pi where I have home assistant running although I have an older raspberry pi 3 that I could use if needed).

Since I don't foresee being able to purchase a new adapter anytime soon, I wondered. I have an airpump (saunier duval) and one heat recuperator (Wolf CWL-300) that use EBUS. Can two devices be connected to a single adapter?

If that isn't possible (it would surprise me if it was), can a single ebusd daemon process work with two adapters or we'd need two ebusd processes with different config files?

**Comment by john30** [comment]
usually multiple ebusses can't be just connected together physically due to powering issues and address conflicts. I have 3 busses at home and I wouldn't dare to connect them.
a single ebusd handles exactly one bus, so you have to run one for each bus to address

**Reply by richardcastle** [reply]
Hi John. Thanks for coming back to me on this. Whilst both instances have a neat distinction via MQTT/HA, I was wondering how I select between instances using the command line tool _ebusctl_. Is there a way to specify which of my two instances should get the command? I'm not seeing it mentioned in the docs unfortunately.

**Reply by docbunsen** [reply]
Hi, I had the same issue.  I started each `ebusd` instance with the `--port` option using a different PORT.
With `ebusctl --port=PORT` you can connect to the correct instance.

**Comment by richardcastle** [comment]
So I've got a configuration as described with two Vailant boilers. I've got two ebuzzz adapters built and connected up with one for each boiler.

At the minute I'm getting all the variables coming through as duplicates in HomeAssistant. Is there a recommended configuration to separate out two identical boilers that are slaved together (not using ebus). Thanks!

**Reply by dk7jb** [reply]
I would be interested to know how you integrated the adapters into HomeAssistant.

**Reply by richardcastle** [reply]
Happy to share configuration. I did find some details in the docs and configuration file examples on how to establish two ebusd instances on my BoilerPi! Each one has it's on configuration file pointing to a specific USB device. I setup different MQTT nodes for each configuration which resulted in separate topics ebusd1, ebusbai1, ebusd2, ebusbai2 surfacing within HA.

**Reply by Infernoken76** [reply]
Hi, 
I have a similar situation where I have a gasboiler Ecotec Vaillant VC126 and a heatpump aroTHERM Plus 75/6...

Both of them work together with an ebus bridge VR32 and right now I have a ebus V5 dongle working with the heatpump.
But it cannot see the gasboiler as it's behind the VR32...

So I bought a second ebus V5 dongle to monitor the gasboiler...

But how do I get this data into home assistant?

I already have the ebusd add-on running for my first dongle (https://github.com/LukasGrebe/ha-addons)...
Do you know how to integrate 2 dongles into home assistant?

**Reply by JonesGW-MGD** [reply]
@Infernoken76 - I have replied to your separate discussion topic regarding accessing your boiler behind a VR32 

https://github.com/john30/ebusd/discussions/1223#discussioncomment-8749250
